x/stakeibc/keeper/msg_server_restore_interchain_account_test.go

```diff
@@ -105,3 +124,66 @@ func (s *KeeperTestSuite) TestRestoreInterchainAccount_FailsIfAccountExists() {
 	)
 	s.Require().EqualError(err, expectedErrMsg, "registered ica account fails when account already exists")
 }
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_RevertDepositRecords_Failure() {
+	tc := s.SetupRestoreInterchainAccount()
+	s.CreateICAChannel("GAIA.DELEGATION")
+	msg := tc.validMsg
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
+	expectedErrMsg := fmt.Sprintf("existing active channel channel-1 for portID icacontroller-%s.DELEGATION on connection %s for owner %s.DELEGATION: active channel already set for this owner",
+		tc.validMsg.ChainId,
+		s.TransferPath.EndpointB.ConnectionID,
+		tc.validMsg.ChainId,
+	)
+	s.Require().EqualError(err, expectedErrMsg, "registered ica account fails when account already exists")
+	// Verify the deposit record state was reverted
+	for i := 0; i < 2; i++ {
+		depositRecord, found := s.App.RecordsKeeper.GetDepositRecord(s.Ctx(), uint64(i))
+		s.Require().True(found, "deposit record found")
+		s.Require().Equal(recordtypes.DepositRecord_DELEGATION_IN_PROGRESS, depositRecord.Status, "deposit record status should NOT nsg be reverted")
+	}
+}
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_NoRecordChange_Success() {
+	tc := s.SetupRestoreInterchainAccount()
+	owner := "GAIA.WITHDRAWAL"
+	channelID := s.CreateICAChannel(owner)
+	portID := icatypes.PortPrefix + owner
+
+	// Confirm there are two channels originally
+	channels := s.App.IBCKeeper.ChannelKeeper.GetAllChannels(s.Ctx())
+	s.Require().Len(channels, 2, "there should be 2 channels initially (transfer + delegate)")
+
+	// Close the delegation channel
+	channel, found := s.App.IBCKeeper.ChannelKeeper.GetChannel(s.Ctx(), portID, channelID)
+	s.Require().True(found, "delegation channel found")
+	channel.State = channeltypes.CLOSED
+	s.App.IBCKeeper.ChannelKeeper.SetChannel(s.Ctx(), portID, channelID, channel)
+
+	// Restore the channel
+	msg := tc.validMsg
+	msg.AccountType = stakeibc.ICAAccountType_WITHDRAWAL
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
+	s.Require().NoError(err, "registered ica account successfully")
+
+	// Confirm the new channel was created
+	channels = s.App.IBCKeeper.ChannelKeeper.GetAllChannels(s.Ctx())
+	s.Require().Len(channels, 3, "there should be 3 channels after restoring")
+
+	// Confirm the new channel is in state INIT
+	newChannelActive := false
+	for _, channel := range channels {
+		// The new channel should have the same port, a new channel ID and be in state INIT
+		if channel.PortId == portID && channel.ChannelId != channelID && channel.State == channeltypes.INIT {
+			newChannelActive = true
+		}
+	}
+	s.Require().True(newChannelActive, "a new channel should have been created")
+
+	// Verify the deposit record state was reverted
```

Comment:
*Verify the deposit record state was NOT reverted, because a non-delegation channel was restored